roofit/roofitcore/test/AutomaticDiff/simple.cpp

```diff
@@ -0,0 +1,32 @@
+#include <RooRealVar.h>
+#include <RooWorkspace.h>
+#include <RooRealProxy.h>
+#include <RooGaussian.h>
+#include <RooDataSet.h>
+#include <RooAbsPdf.h>
+#include <TROOT.h>
+#include <TSystem.h>
+#include <RooFitResult.h>
+#include <RooListProxy.h>
+#include <TMath.h>
+
+#include "gtest/gtest.h"
+
+class Interface : public ::testing::Test {
+};
+
+TEST(Interface, createNLLRooAbsGradFuncWrapper)
+{
+
+   using namespace RooFit;
+
+   RooRealVar x("x", "x", 0, -10, 10);
+   RooRealVar mu("mu", "mu", 0, -10, 10);
+   RooRealVar sigma("sigma", "sigma", 2.0, 0.01, 10);
+
+   RooGaussian gauss{"gauss", "gauss", x, mu, sigma};
+
+   std::unique_ptr<RooDataSet> data{gauss.generate(x, 10)};
+
+   auto nll = gauss.createNLL(*data, CodeSquashing(true));
```

Comment:
Can we store to a `std::string` the produced code, call `gInterpreter->ProcessLine` and assert if we fail to compile it?

Reply:
maybe we can do it in the `createNLL` function itself?